service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -496,12 +496,23 @@ fn generate_struct_fields<P: GenerateProtocol>(service: &Service,
                                     protocol_generator.timestamp_type());
         let name = generate_field_name(member_name);
 
-        if shape.required(member_name) {
-            lines.push(format!("pub {}: {},", name, rs_type))
-        } else if name == "type" {
-            lines.push(format!("pub aws_{}: Option<{}>,", name,rs_type))
+        // For structs that can contain another of themselves, we need to box them.
```

Comment:
Surely there's a better way of making this work. 😄 I've also not tested it so it probably doesn't work as is, but it does compile.